linera-storage/src/migration.rs

```diff
@@ -0,0 +1,572 @@
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+        BaseKey::BlobState(blob_id) => {
+            let root_key = RootKey::Blob(blob_id).bytes();
+            Ok((root_key, ONE_KEY.to_vec()))
+        }
+        BaseKey::Event(event_id) => {
+            let root_key = RootKey::Event(event_id.chain_id).bytes();
+            let key = to_event_key(&event_id);
+            Ok((root_key, key))
+        }
+        BaseKey::BlockExporterState(index) => {
+            let root_key = RootKey::BlockExporterState(index).bytes();
+            Ok((root_key, UNUSED_EMPTY_KEY.to_vec()))
+        }
+        BaseKey::NetworkDescription => {
+            let root_key = RootKey::NetworkDescription.bytes();
+            Ok((root_key, DEFAULT_KEY.to_vec()))
+        }
+    }
+}
+
+impl<Database, C> DbStorage<Database, C>
+where
+    Database: KeyValueDatabase + Clone + Send + Sync + 'static,
+    Database::Store: KeyValueStore + Clone + Send + Sync + 'static,
+    C: Clock + Clone + Send + Sync + 'static,
+    Database::Error: From<bcs::Error> + Send + Sync,
+{
+    async fn migrate_shared_partition(
+        &self,
+        first_byte: &u8,
+        keys: Vec<Vec<u8>>,
+    ) -> Result<(), ViewError> {
+        tracing::info!(
+            "migrate_shared_partition for {} keys starting with with {first_byte}",
+            keys.len()
+        );
+        for (index, chunk_keys) in keys.chunks(BLOCK_KEY_SIZE).enumerate() {
+            tracing::info!("processing chunk {index} of size {}", chunk_keys.len());
+            let chunk_base_keys = chunk_keys
+                .iter()
+                .map(|key| {
+                    let mut base_key = vec![*first_byte];
+                    base_key.extend(key);
+                    base_key
+                })
+                .collect::<Vec<Vec<u8>>>();
+            let store = self.database.open_shared(&[])?;
+            let values = store
+                .read_multi_values_bytes(chunk_base_keys.to_vec())
+                .await?;
+            let mut batch = MultiPartitionBatch::new();
+            for (base_key, value) in chunk_base_keys.iter().zip(values) {
+                let value = value.ok_or(ViewError::MissingEntries)?;
+                tracing::info!("base_key={base_key:?} value={value:?}");
+                let (root_key, key) = map_base_key(base_key)?;
+                batch.put_key_value_bytes(root_key, key, value);
+            }
+            self.write_batch(batch).await?;
+            // Now delete the keys
+            let mut batch = Batch::new();
+            for key in chunk_base_keys {
+                batch.delete_key(key.to_vec());
+            }
+            store.write_batch(batch).await?;
+        }
+        Ok(())
+    }
+
+    async fn migrate_v0_to_v1(&self) -> Result<(), ViewError> {
+        for first_byte in MOVABLE_KEYS_0_1 {
+            let store = self.database.open_shared(&[])?;
+            let keys = store.find_keys_by_prefix(&[*first_byte]).await?;
```

Comment:
(This will load all keys of each type into memory at the same time; that should still be fine, should it, even for the testnet validators? I.e. loading _all_ block hashes, or loading _all_ blob IDs?)